streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -94,39 +93,43 @@ public InternalStreamsBuilder(final InternalTopologyBuilder internalTopologyBuil
                                      final ConsumedInternal<K, V> consumed,
                                      final MaterializedInternal<K, V, KeyValueStore<Bytes, byte[]>> materialized) {
         final StoreBuilder<KeyValueStore<K, V>> storeBuilder = new KeyValueStoreMaterializer<>(materialized)
-                .materialize();
+            .materialize();
 
         final String source = newProcessorName(KStreamImpl.SOURCE_NAME);
         final String name = newProcessorName(KTableImpl.SOURCE_NAME);
+
+        StatefulSourceNode.StatefulSourceNodeBuilder<K, V> statefulSourceNodeBuilder = StatefulSourceNode.statefulSourceNodeBuilder();
+
         final KTable<K, V> kTable = createKTable(consumed,
-                                                 topic,
                                                  storeBuilder.name(),
                                                  materialized.isQueryable(),
                                                  source,
-                                                 name);
+                                                 name,
+                                                 statefulSourceNodeBuilder);
```

Comment:
Instead of passing in the `statefulSourceNodeBuilder` to `createKTable`, we can just get `statefulSourceNodeBuilder.withProcessorParameters(new ProcessorParameters<>(new KTableSource<>(storeName), name))` here. This way there is less code leaking.

Ditto elsewhere.

Reply:
Done.  I was able to remove the method `createKTable` as it was used only in place.  In other places, we need to pass in a builder as there are many required bits of information generated that need to get captured.  The `KTable.buildJoin`  method is a good example.